waltz-test/src/main/python/waltz_ducktape/tests/validation/connection_interruption_test.py

```diff
@@ -0,0 +1,194 @@
+        :param num_interruptions: Number of connection interruption cycles
+        :param delay_between_interruptions: Interval in seconds that represents duration between network interruptions
+        :param num_active_partitions: Number of active partitions
+        :param processing_duration: Time in seconds within which it is safe to assume that processed transactions gets stored
+        """
+
+        partition = randrange(num_active_partitions)
+        node_idx = self.get_server_node_idx(partition)
+
+        # Start waltz cluster and wait until a storage node registers first transaction
+        self.verifiable_client.start(validation_cmd)
+        admin_port = self.waltz_storage.admin_port
+        port = self.waltz_storage.port
+        storage = self.get_host(self.waltz_storage.nodes[0].account.ssh_hostname, admin_port)
+        wait_until(lambda: self.is_max_transaction_id_updated(storage, port, partition, -1), timeout_sec=timeout)
+
+        node = self.waltz_server.nodes[node_idx]
+        for interruption in range(num_interruptions):
+            sleep(delay_between_interruptions)
+            try:
+                # disable connection on port
+                self.drop_traffic_to_port(node, self.waltz_server.port)
+
+                # verify that during network interruption number of stored transactions didn't increase
+                # sleep time to assure that transactions from server are propagated to storage nodes
+                sleep(processing_duration)
+                cur_high_watermark = self.get_storage_max_transaction_id(storage, port, partition)
+                sleep(max(interrupt_duration - processing_duration, 0))
+                assert not self.is_max_transaction_id_updated(storage, port, partition, cur_high_watermark), \
+                    'Network interruption failed, newly stored transactions detected'
+
+                # enable connection on port
+                self.enable_traffic_to_port(node, self.waltz_server.port)
+            finally:
+                """
+                delete the added iptable rule as it is not removed with the end of waltz process and could persist on 
+                waltz-server VM, if process is signaled to end (^C) or an exception is thrown during execution of the try block. 
+                If iptable rule is not present (already deleted) nothing happens on the waltz-server VM side.
+                """
+                self.enable_traffic_to_port(node, self.waltz_server.port)
+
+        wait_until(lambda: self.verifiable_client.task_complete() == True, timeout_sec=timeout,
+                   err_msg="verifiable_client failed to complete task in %d seconds." % timeout)
+
+    class StorageNodeInfo:
+        """
+        Representation of a storage node and number of transactions stored on a node
+        """
+        def __init__(self, node):
+            self.node = node
+            self.total_number_of_transactions_across_partitions = 0
```

Comment:
I am assuming `total_number_of_transactions_across_partitions` stores total number of existing transactions (across active partition) on the storage node.
If yes, then better to rename it to be more intuitive. 
`total_number_of_existing_transactions_across_partitions` may be?